### PR TITLE
Docs for configuring Lambda function settings

### PR DESCRIPTION
**Pull request by dabit3**
This adds a new guide showing how to configure AWS Lambda configurations using the Amplify CLI

### CODE REVIEW

**Review by medelman17 — APPROVED** [review]

File: docs/guides/functions/configuring-lambda.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+title: Configuring Lambda function settings
+description: How to configure custom settings for your Lambda function
+---
+
+In this guide you will learn how to configure custom settings for a Lambda function.
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:6]:
```suggestion
Learn how to configure custom settings (including runtime, memory, environment variables, and secrets) for a Lambda function.
```

File: docs/guides/functions/configuring-lambda.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+### Overview
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:8]: this title is not really needed, i think.

File: docs/guides/functions/configuring-lambda.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+When creating a function within an Amplify project, you are given several options to choose how you would like your function to be configured. You may find that your application requires a different configuration than the CLI set up for you.
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:10]:
```suggestion
You may want to override the Amplify CLI default configurations for your Lambda function or configure changes not available within the `amplify add function` workflow.
```
Maybe something like this to keep it shorter. Then you can collapse all of this with the primary sentence on top.

File: docs/guides/functions/configuring-lambda.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+For example, when creating a `Node.js` function, the CLI will configure a version for you, also setting a default memory size among other things. There are a few things you may want to override or configure:
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:12]:
```suggestion
*Example*: When creating a `Node.js` function, the CLI will automatically configure a runtime version, a default memory size, and more. There are a few things you may want to override or configure:
```

File: docs/guides/functions/configuring-lambda.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+1. Runtime
+2. Memory size
+3. Environment variables
+
+Let's look at how to update all of these things.
+
+### Updating the Runtime
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:20]: Use H2

File: docs/guides/functions/configuring-lambda.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+Depending on what version you are using, the CLI will set a default runtime version. You may want to tweak the version of the runtime to be either a newer or older version.
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:22]:
```suggestion
You may want to tweak the runtime version to be either a newer or older version than the Amplify-generated default.
```

File: docs/guides/functions/configuring-lambda.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+Let's say we've deployed a Lambda function using a Node.js runtime and we want to modify the version of the runtime to be `14.x`.
+
+To do so, open __amplify/backend/function/function-name/function-name-cloudformation-template.json__ and set the `RunTime` property in the `LambdaFunction` resource:
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:26]:
```suggestion
To do so, open __amplify/backend/function/function-name/function-name-cloudformation-template.json__ and set the `Runtime` property in the `LambdaFunction` resource to:
```

File: docs/guides/functions/configuring-lambda.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+}
+```
+
+Next, deploy the updates using the Amplify CLI:
+
+```sh
+amplify push
+```
+
+### Updating the default memory size
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:46]: H2

File: docs/guides/functions/configuring-lambda.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+Next, deploy the updates using the Amplify CLI:
+
+```sh
+amplify push
+```
+
+_To learn more about optimizing resources allocation for Lambda functions, check out [this](https://dev.to/aws/deep-dive-finding-the-optimal-resources-allocation-for-your-lambda-functions-35a6) blog post._
+
+
+### Setting an environment variable
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:74]: H2

File: docs/guides/functions/configuring-lambda.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+A very common scenario is the need to set and use an environment variable in your Lambda function.
+
+There are two main approaches (other than manually adding values in the Lambda console):
+
+1. If your value is secret, you can use [Secrets Manager](https://aws.amazon.com/secrets-manager/).
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:80]:
```suggestion
### 1. Configuring secret values

If your value is secret, you can use [Secrets Manager](https://aws.amazon.com/secrets-manager/).
```

File: docs/guides/functions/configuring-lambda.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+```js
+const AWS = require('aws-sdk')
+
+const secretsManager = new AWS.SecretsManager()
+const secret = await secretsManager.getSecretValue({ SecretId: 'YOUR_KEY' }).promise()
+
+console.log(secret.SecretString)
+```
+
+2. If your value is just a configuration value you can configure the CloudFormation configuration locally to set the value - in __amplify/backend/function/function-name/function-name-cloudformation-template.json__
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:119]:
```suggestion
### 2. Configuring non-secret values

If your value is just a configuration value, you can configure the CloudFormation configuration locally to set the value - in __amplify/backend/function/function-name/function-name-cloudformation-template.json__
```

File: docs/guides/functions/configuring-lambda.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+```
+
+_To learn more about optimizing resources allocation for Lambda functions, check out [this](https://dev.to/aws/deep-dive-finding-the-optimal-resources-allocation-for-your-lambda-functions-35a6) blog post._
+
+
+### Setting an environment variable
+
+A very common scenario is the need to set and use an environment variable in your Lambda function.
+
+There are two main approaches (other than manually adding values in the Lambda console):
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:78]:
```suggestion
There are generally two types of environment variables:
- Secret values (example: access keys, API keys etc.)
- Non-secret values (example: endpoint information, locale information etc.)
```

File: docs/guides/functions/configuring-lambda.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+const secretsManager = new AWS.SecretsManager()
+const secret = await secretsManager.getSecretValue({ SecretId: 'YOUR_KEY' }).promise()
+
+console.log(secret.SecretString)
+```
+
+2. If your value is just a configuration value you can configure the CloudFormation configuration locally to set the value - in __amplify/backend/function/function-name/function-name-cloudformation-template.json__
+
+For this purpose there is a section in the template - `Parameters` - that you can set.
```
Comment (renebrandel) [inline docs/guides/functions/configuring-lambda.md:121]:
```suggestion
For this purpose, there is a section in the template - `Parameters` - that you can set.
```

**Review by renebrandel — APPROVED** [review]
Content is great. Suggestions for conciseness and proper structure.

**Review by kaustavghosh06 — APPROVED** [review]

**Review by kaustavghosh06 — APPROVED** [review]

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**